docs/utilities/data_classes.md

```diff
@@ -81,6 +82,45 @@ Event Source | Data_class
     The examples provided below are far from exhaustive - the data classes themselves are designed to provide a form of
     documentation inherently (via autocompletion, types and docstrings).
 
+### API Gateway Authorizer V2
+
+> New in 1.20.0
+
+It is used for API Gateway HTTP API lambda authorizer payload version 2. See blog post
+[Introducing IAM and Lambda authorizers for Amazon API Gateway HTTP APIs](https://aws.amazon.com/blogs/compute/introducing-iam-and-lambda-authorizers-for-amazon-api-gateway-http-apis/){target="_blank"}
+or [Working with AWS Lambda authorizers for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-lambda-authorizer.html){target="_blank"}
+for more details
+
+Below is a simple example of an HTTP API lambda authorizer looking up user details by `x-token` header and using
+`APIGatewayAuthorizerResponseV2` to return the declined response when user is not found or authorized and include
+the user details in the request context.
+
+=== "app.py"
+
+    ```python
+    from aws_lambda_powertools.utilities.data_classes import event_source
+    from aws_lambda_powertools.utilities.data_classes.api_gateway_authorizer_event import (
+        APIGatewayAuthorizerEventV2,
+        APIGatewayAuthorizerResponseV2,
+    )
+
+
+    def get_user_by_token(token):
+        ...
```

Comment:
let's strive to make a working example

Reply:
@heitorlessa  i am a fake, but working one.